### PR TITLE
Ready button implementation.

### PR DESCRIPTION
**Pull request by admoore0**

### CODE REVIEW

File: assets/icons/checkmark.png
Comment (thearst3rd) [inline assets/icons/checkmark.png:1]: ✅

File: src/autoload/multiplayer.gd [hunk 1]
```diff
@@ -74,7 +75,8 @@ class GameInfo:
 	func serialize() -> Dictionary:
 		var output := {
 			"server_name": server_name,
-			"mode": int(mode)
+			"mode": int(mode),
+			"max_players": max_players
```
Comment (thearst3rd) [inline src/autoload/multiplayer.gd]: gonna merge conflict lol

Comment (thearst3rd) [reply src/autoload/multiplayer.gd]: (and also add trailing comma plz)

**Review by kdreese — DISMISSED** [review]

File: src/states/menus/lobby.gd [hunk 2]
```diff
@@ -68,6 +73,22 @@ func show_menu() -> void:
 func sync_colors(_chosen_colors: Dictionary):
 	chosen_colors = _chosen_colors
 
+
+# Called on the server when someone says they are ready.
+@rpc("any_peer", "call_local")
+func player_ready(player_id: int, state: bool):
+	players_ready[player_id] = state
+	sync_ready_status.rpc(players_ready)
+	update_display()
```
Comment (thearst3rd) [inline src/states/menus/lobby.gd]: No need for player_id
```suggestion
func player_ready(state: bool):
	players_ready[multiplayer.get_remote_sender_id()] = state
	sync_ready_status.rpc(players_ready)
	update_display()
```

File: src/states/menus/lobby.gd [hunk 3]
```diff
@@ -134,6 +155,12 @@ func on_color_button_press(idx: int) -> void:
 	chosen_colors[get_multiplayer().get_unique_id()] = idx
 	sync_colors.rpc(chosen_colors)
 	update_display.rpc()
+	if not %ReadyPanel.visible:
+		%ReadyPanel.show()
+
+
+func on_ready_checkbox_update(state: bool) -> void:
+	player_ready.rpc_id(1, get_multiplayer().get_unique_id(), state)
```
Comment (thearst3rd) [inline src/states/menus/lobby.gd]: Correspondingly
```suggestion
	player_ready.rpc_id(1, state)
```

File: src/states/menus/lobby.gd [hunk 4]
```diff
@@ -35,11 +33,15 @@ const NUM_ROWS = 8
 @onready var start_button: Button = %StartButton
 @onready var mode_drop_down: MenuButton = %ModeDropDown
 @onready var ping_timer: Timer = %PingTimer
+@onready var player_table_row: PanelContainer = %Row.duplicate()
```
Comment (thearst3rd) [inline src/states/menus/lobby.gd]: Maybe name `player_table_row_template`?

**Review by thearst3rd — CHANGES_REQUESTED** [review]

**Review by thearst3rd — APPROVED** [review]

**Pull request merged**